components/support/ktx/src/main/java/mozilla/components/support/ktx/android/view/View.kt

```diff
@@ -100,6 +106,39 @@ fun View.setPadding(padding: Padding) {
     }
 }
 
+/**
+ * Creates a [CoroutineScope] that is active as long as this [View] is attached. Once this [View]
+ * gets detached this [CoroutineScope] gets cancelled automatically.
+ *
+ * By default coroutines dispatched on the created [CoroutineScope] run on the main dispatcher.
+ *
+ * @throws IllegalArgumentException if this [View] is not attached yet. It is the responsibility of
+ * the caller to make sure the [View] is attached before turning it into a [CoroutineScope], e.g.
+ * inside a [Fragment] call this from [Fragment.onViewCreated].
+ */
+@MainThread
+fun View.toScope(): CoroutineScope {
+    if (!isAttachedToWindow) {
+        // We can't pause a CoroutineScope until the View is attached; So we just throw in this
+        // situation. The calling code needs to make sure the View is attached first (e.g. call
+        // this from onViewCreated in a Fragment).
+        throw IllegalStateException("View is not attached")
```

Comment:
After applying the change to the SearchFragment in Fenix (in `onViewCreated`) we're crashing here now:
```kotlin
consumeFrom(searchStore) { state ->            
  awesomeBarView.update(state)
  toolbarView.update(state)
  updateSearchEngineIcon(state)
  updateSearchShortuctsIcon(state)
  updateSearchWithLabel(state)
}
```

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.fenix.debug/org.mozilla.fenix.HomeActivity}: java.lang.IllegalStateException: View is not attached
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2666)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2727)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4520)
        at android.app.ActivityThread.-wrap19(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1484)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6121)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
     Caused by: java.lang.IllegalStateException: View is not attached
        at mozilla.components.support.ktx.android.view.ViewKt.toScope(View.kt:125)
        at mozilla.components.lib.state.ext.FragmentKt.consumeFrom(Fragment.kt:33)
        at org.mozilla.fenix.search.SearchFragment.onViewCreated(SearchFragment.kt:170)
```

Reply:
This is triggered by the same flow we tested before:
- trigger a search
- settings -> private tab

Wondering if we should just have a `broadcastChannel` method that takes a view and uses a `SubscriptionViewBinding`?

Reply:
This is surprising. I tested this multiple times and looked at the flow and it never crashed here. I'll try again today.

> Wondering if we should also have a broadcastChannel method that takes a view and uses a SubscriptionViewBinding?

Technically this shouldn't be needed: The owner is used to pause/resume which never happens for the View (it only gets attached/detached). And the scope should get cancelled once the `View` detaches causing nothing to get consumed anymore.